app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/UniqueDatabaseManagement.kt

```diff
@@ -54,50 +69,76 @@ class UniqueDatabaseManagement constructor(
         )
     }
 
-    suspend fun downloadDatabase(databaseName: String): DatabaseManagement {
-        val firestoreDbManagement =
-            DefaultDatabaseManagement(FirestoreDatabaseService(databaseName, firestore))
-        val pictureRepository = PictureRepository(databaseName, context)
+    fun downloadDatabase(databaseName: String): Deferred<DefaultDatabaseManagement> =
+        CoroutineScope(Dispatchers.IO).async {
```

Comment:
```suggestion
    suspend fun downloadDatabase(databaseName: String): DatabaseManagement =
        withContext(Dispatchers.IO) {
```
This is more idiomatic, the Deferred seems unnecessary.

Reply:
Indeed